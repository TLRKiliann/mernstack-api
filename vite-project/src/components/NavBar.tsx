import { Component } from 'react'
import { Link } from 'react-router-dom'
import koalatree from '../assets/koala_tree.png'
import './styleComponents/NavBar.scss'


export default class NavBar extends Component {
  render() {
    return(
      <div data-test-id="testId" className="div--headernav">
        <nav >

          <div className="div--img">
            <img src={koalatree} width='53px' height='58px' alt='no img navbar'/>
          </div>

          <ul>
            <li>
              <Link
                className="link--tagnav"
                to='/'
              >
                Home
              </Link>
            </li>
            <li>
              <Link
                className="link--tagnav"
                to='/services'
              >
                Services
              </Link>
            </li>
            <li>
              <Link
                className="link--tagnav"
                to='/help'
              >
                Help
              </Link>
            </li>
          </ul>
        </nav>
      </div>
    )
  }
}