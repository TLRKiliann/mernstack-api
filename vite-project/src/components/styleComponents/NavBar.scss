$lightgrey_54: #545454;
$lightgrey_50: #505050;
$lightgrey_48: #484848;
$lightgrey_42: #424242;
$lightgrey_38: #383838;
$slategray: #708090;
$darkmagenta: #8B008B;
$sandybrown: #f4a460;
$lightgreen: #90ee90;
$whitesmoke: #f5f5f5;
$font_junction: 'Junction', sans-serif;
$font_allura: 'Allura', cursive;

.div--headernav {
  position: absolute;
  top: 0px;
  right: 0px;
  left: 0px;
  width: 100%;
  height: 90px;
  margin: auto;
  display: flex;
  flex-direction: column;
  background: linear-gradient(90deg, rgba(244,164,96, 1), rgba(195,87,116, 1));
  box-shadow: 0px 0px 7px rgba(244,164,96, 1);
  z-index: 2;
  .navbar--nav {
    width: 100%;
    margin: auto;
    display: flex;
    .div--imgnavbar {
      position: relative;
      margin: auto;
      left: 20px;
      display: flex;
      background: linear-gradient(180deg,
        rgba(195,87,116, 1), rgba(244,164,96, 1));
      border: none;
      outline: none;
      border-radius: 50%;
      z-index: 2;
      .img--koala {
        width: 57px;
        height: 60px;
      }
    }
    .ul--navbar {
      padding-left: 40px;
      width: 100%;
      margin: auto;
      display: flex;
      flex-direction: row;
      justify-content: flex-end;
      .link--tagnav {
        text-decoration: none;
        text-shadow: 1px 0px 1px #000;
        color: $whitesmoke;
        &:hover {
         color: rgba(244,164,96, 1);
         text-shadow: 1px 0px 1px rgba(244,164,96, 1);
        }
      }
    }
  }
}
.div--headernav > nav > ul > li {
  margin-right: 30px;
  padding: 20px 0px;
  display: flex;
  align-items: center;
  justify-content: space-around;
  font-family: $font_allura;
  font-size: 2.4rem;
  font-weight: bold;
}
.div--userloginnavbar {
  width: auto;
  height: 50px;
  margin: auto;
  margin-right: 10px;
  display: flex;
  flex-direction: row;
  background: linear-gradient(180deg,
    rgba(195,87,116, 1), rgba(244,164,96, 1));
  border: 1mm inset rgba(195,87,116, 1);
  border-radius: 15px;
  .div--userplace {
    margin-left: 10px;
    width: auto;
    display: flex;
    align-items: center;
    justify-content: center;
    .span--navbarusername {
      padding: 5px 10px;
      font-family: $font_junction;
      font-size: 0.9rem;
      font-weight: bold;
      background-color: $lightgrey_38;
      border: 1mm outset $lightgrey_50;
      border-radius: 7px;
      color: rgba(244,164,96, 1);
    }
    .span--usrnavbar {
      padding: 5px 10px;
      background-color: $lightgrey_38;
      border-radius: 5px;
      color: $lightgreen;
    }
  }
  .div--login {
    margin-right: 10px;
    margin-left: 10px;
    display: flex;
    align-items: center;
    justify-content: center;
    .link--logintag {
      padding: 5px 10px;
      display: flex;
      margin: auto;
      text-decoration: none;
      font-family: 'Junction', sans-serif;
      font-size: 0.9rem;
      background-color: $lightgrey_38;
      border: 1mm outset $lightgrey_50;
      border-radius: 10px;
      color: $whitesmoke;
      &:hover {
        background: $lightgrey_50;
        border: 1mm outset $lightgrey_54;
      }
      &:active {
        transform: scale(0.95);
        background-color: $lightgrey_38;
        border: 1mm inset $lightgrey_50;
        color: $lightgreen;
      }
    }
  }
  .link--logout {
    padding: 5px 10px;
    display: flex;
    align-items: center;
    text-decoration: none;
    font-family: $font_junction;
    font-size: 0.9rem;
    background-color: $lightgrey_38;
    border: 1mm outset $lightgrey_50;
    border-radius: 10px;
    color: $whitesmoke;
    &:hover {
      transform: scale(1);
      background-color: $lightgrey_50;
      border: 1mm outset $lightgrey_50;
      color: $whitesmoke;
    }
    &:active {
      transform: scale(0.9);
      background-color: $lightgrey_38;
      border: 1mm inset $lightgrey_42;
      color: rgba(244,164,96, 1);
    }
  }
}

@media only screen and (min-width: 360px) and (max-width: 768px) {
  .div--headernav {
    position: absolute;
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    .navbar--nav {
      width: 100%;
      margin: auto;
      display: flex;
      .div--imgnavbar {
        position: absolute;
        top: 20px;
        width: 40px;
        height: 40px;
        display: flex;
        align-items: center;
        justify-content: center;
        .img--koala {
          width: 22px;
          height: 22px;
        }
      }
      .ul--navbar {
        padding-left: 10px;
        width: 100%;
        margin: auto;
        display: flex;
        flex-direction: row;
        justify-content: flex-end;
      }
    }
  }
  .div--headernav > nav > ul > li {
    margin-right: 20px;
    font-size: 1.2rem;
  }
  .div--userloginnavbar {
    width: auto;
    height: 40px;
    margin: auto;
    margin-right: 20px;
    display: flex;
    flex-direction: row;
    .div--userplace {
      margin-left: 5px;
      width: auto;
      display: flex;
      align-items: center;
      justify-content: center;
      .span--navbarusername {
        padding: 3px 5px;
        font-family: $font_junction;
        font-size: 0.6rem;
        font-weight: bold;
      }
      .span--usrnavbar {
        padding: 3px 5px;
      }
    }
    .div--login {
      margin-right: 5px;
      margin-left: 5px;
      display: flex;
      align-items: center;
      justify-content: center;
      .link--logintag {
        padding: 6px 6px;
        display: flex;
        margin: auto;
        text-decoration: none;
        font-family: $font_junction;
        font-size: 0.6rem;
      }
    }
    .link--logout {
      padding: 3px 5px;
      display: flex;
      align-items: center;
      text-decoration: none;
      font-family: $font_junction;
      font-size: 0.6rem;      
    }
  }
}