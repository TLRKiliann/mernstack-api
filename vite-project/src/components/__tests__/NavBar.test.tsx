import React from "react"
import { screen, render } from '@testing-library/react'
import { test, expect} from 'vitest'
import { MemoryRouter } from 'react-router-dom'
import '@testing-library/jest-dom'
import { create } from 'react-test-renderer'
import NavBar from "../NavBar.tsx"

test('NavBar renders by test id', () => {
  render(
    <MemoryRouter>
      <NavBar />
    </MemoryRouter>
  )
  const linkElement = screen.getByTestId("testId")
  expect(linkElement).toBeInTheDocument()
})

test('NavBar toMatchSnapshot', () => {
  const treeNav = create(
    <MemoryRouter>
      <NavBar />
    </MemoryRouter>
  )
  expect(treeNav.toJSON()).toMatchSnapshot()
})