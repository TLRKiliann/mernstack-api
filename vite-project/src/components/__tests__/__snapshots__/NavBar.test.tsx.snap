// Vitest Snapshot v1

exports[`NavBar toMatchSnapshot 1`] = `
<div
  className="div--headernav"
  data-testid="testId"
>
  <nav
    className="navbar--nav"
  >
    <div
      className="div--imgnavbar"
    >
      <img
        alt="no img navbar"
        className="img--koala"
        height="70px"
        src="/src/assets/logo/koala_tree.png"
        width="65px"
      />
    </div>
    <ul
      className="ul--navbar"
    >
      <li>
        <a
          className="link--tagnav"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className="link--tagnav"
          href="/profile"
          onClick={[Function]}
        >
          Profile
        </a>
      </li>
      <li>
        <a
          className="link--tagnav"
          href="/online"
          onClick={[Function]}
        >
          Online
        </a>
      </li>
    </ul>
    <div
      className="div--userloginnavbar"
    >
      <div
        className="div--userplace"
      >
        <span
          className="span--usrnavbar"
        >
          â†¦
        </span>
      </div>
      <div
        className="div--login"
      >
        <span
          className="span--navbarlogin"
        >
          <a
            className="link--logout"
            href="/login"
            onClick={[Function]}
          >
            Logout
          </a>
        </span>
      </div>
    </div>
  </nav>
</div>
`;
